base_layer/core/src/chain_storage/accumulated_data.rs

```diff
@@ -182,84 +187,75 @@ impl<'de> Visitor<'de> for DeletedBitmapVisitor {
+        Self {
+            previous_accum,
+            hash: None,
+            current_total_kernel_offset: None,
+            current_achieved_target: None,
+        }
     }
+}
 
-    pub fn total_kernel_offset(
-        mut self,
-        previous_kernel_offset: &BlindingFactor,
-        current_offset: &BlindingFactor,
-    ) -> Self
-    {
-        self.total_kernel_offset = Some(previous_kernel_offset + current_offset);
+impl BlockHeaderAccumulatedDataBuilder<'_> {
+    pub fn with_hash(mut self, hash: HashOutput) -> Self {
+        self.hash = Some(hash);
         self
     }
 
-    pub fn target_difficulty(mut self, target: Difficulty) -> Self {
-        self.target_difficulty = Some(target);
+    pub fn with_total_kernel_offset(mut self, current_offset: BlindingFactor) -> Self {
+        self.current_total_kernel_offset = Some(current_offset);
         self
     }
 
-    pub fn achieved_difficulty(
-        mut self,
-        previous: &BlockHeaderAccumulatedData,
-        algo: PowAlgorithm,
-        achieved: Difficulty,
-    ) -> Self
-    {
-        match algo {
-            PowAlgorithm::Monero => {
-                self.accumulated_monero_difficulty = Some(previous.accumulated_monero_difficulty + achieved);
-                self.accumulated_blake_difficulty = Some(previous.accumulated_blake_difficulty);
-            },
-            PowAlgorithm::Blake => unimplemented!(),
-            PowAlgorithm::Sha3 => {
-                self.accumulated_monero_difficulty = Some(previous.accumulated_monero_difficulty);
-                self.accumulated_blake_difficulty = Some(previous.accumulated_blake_difficulty + achieved);
-            },
-        }
-        self.achieved_difficulty = Some(achieved);
+    pub fn with_achieved_target_difficulty(mut self, achieved_target: AchievedTargetDifficulty) -> Self {
+        self.current_achieved_target = Some(achieved_target);
         self
     }
 
     pub fn build(self) -> Result<BlockHeaderAccumulatedData, ChainStorageError> {
-        let monero_diff = self
-            .accumulated_monero_difficulty
-            .ok_or_else(|| ChainStorageError::InvalidOperation("difficulty not provided".to_string()))?;
+        let previous_accum = self.previous_accum;
+        let hash = self
+            .hash
+            .ok_or_else(|| ChainStorageError::InvalidOperation("hash not provided".to_string()))?;
+
+        let achieved_target = self.current_achieved_target.ok_or_else(|| {
+            ChainStorageError::InvalidOperation("Current achieved difficulty not provided".to_string())
+        })?;
+
+        let (monero_diff, blake_diff) = match achieved_target.pow_algo() {
+            PowAlgorithm::Monero => (
+                previous_accum.accumulated_monero_difficulty + achieved_target.achieved(),
+                previous_accum.accumulated_blake_difficulty,
+            ),
+            PowAlgorithm::Blake => unimplemented!(),
```

Comment:
why do we still have this? Is this used?

Reply:
Because removing it breaks the db - it should have been removed last reset but wasn't - it's already marked with the #testnetreset tag for next time :)